src/c-writer.cc

```diff
@@ -2505,6 +2549,12 @@ void CWriter::WriteExports(CWriterPhase kind) {
           Write("instance");
         }
         WriteParamSymbols(index_to_name);
+        if (IsSingleUnsharedMemory()) {
+          UndoSwapSegueBase();
```

Comment:
Would it make sense to call this `RestoreSegueBase`? 

I guess then the `SwapSequeBase` function would be called `SetSegueBase` but that might no reflect that fact that we cache the old value.  Maybe `InstallSegueBase`? 

This is just bikeshedding though..